p2p/exchange_metrics.go

```diff
@@ -74,35 +72,40 @@ func newExchangeMetrics() (m *exchangeMetrics, err error) {
 	if err != nil {
 		return nil, err
 	}
-	m.trackedPeersNumReg, err = meter.RegisterCallback(m.observeTrackedPeers, m.trackedPeersNumInst)
-	if err != nil {
-		return nil, err
-	}
 	m.disconnectedPeersNumInst, err = meter.Int64ObservableGauge(
 		"hdr_p2p_exch_clnt_disconn_peer_num_gauge",
 		metric.WithDescription("exchange client tracked disconnected peers number"),
 	)
 	if err != nil {
 		return nil, err
 	}
-	m.disconnectedPeersNumReg, err = meter.RegisterCallback(m.observeDisconnectedPeers, m.disconnectedPeersNumInst)
-	if err != nil {
-		return nil, err
-	}
 	m.blockedPeersNumInst, err = meter.Int64ObservableGauge(
 		"hdr_p2p_exch_clnt_block_peer_num_gauge",
 		metric.WithDescription("exchange client blocked peers number"),
 	)
 	if err != nil {
 		return nil, err
 	}
-	m.blockedPeersNumReg, err = meter.RegisterCallback(m.observeBlockedPeers, m.blockedPeersNumInst)
+
+	m.clientReg, err = meter.RegisterCallback(
+		m.observeMetrics,
+		m.trackedPeersNumInst,
+		m.disconnectedPeersNumInst,
+		m.blockedPeersNumInst,
+	)
 	if err != nil {
 		return nil, err
 	}
 	return m, nil
 }
 
+func (m *exchangeMetrics) observeMetrics(ctx context.Context, obs metric.Observer) error {
+	_ = m.observeTrackedPeers(ctx, obs)
+	_ = m.observeDisconnectedPeers(ctx, obs)
+	_ = m.observeBlockedPeers(ctx, obs)
+	return nil
+}
```

Comment:
nit: I think we can now combine the body of all those into a single method. They were made separate assuming multiple callbacks can be registered, but if not lets just keep them coupled.

Reply:
removed methods